risc0/zkvm/src/exec/mod.rs

```diff
@@ -143,19 +155,55 @@ impl<'a> Executor<'a> {
             segment_cycle: init_cycles,
             segments: Vec::new(),
             insn_counter: 0,
+            bonsai_proof_id: bonsai_proof_id,
         }
     }
 
     /// Construct a new [Executor] from an ELF binary.
     pub fn from_elf(env: ExecutorEnv<'a>, elf: &[u8]) -> Result<Self> {
         let program = Program::load_elf(&elf, MEM_SIZE as u32)?;
         let image = MemoryImage::new(&program, PAGE_SIZE as u32)?;
-        Ok(Self::new(env, image, program.entry))
+        let mut bonsai_proof_id: Option<i64> = None;
+
+        // TODO: move this to the run() function once we switch the ELF registration to
+        // use MemoryImages.
+        if std::env::var("BONSAI_DOGFOOD_URL").is_ok() {
```

Comment:
Here you can use a `if let Some(...) = ....` pattern: https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html